selftests/unit/test_utils_iso9660.py

```diff
@@ -0,0 +1,119 @@
+"""
+Verifies the avocado.utils.iso9660 functionality
+"""
+import os
+import shutil
+import sys
+import tempfile
+
+from avocado.utils import iso9660, process
+
+
+if sys.version_info[:2] == (2, 6):
+    import unittest2 as unittest
+else:
+    import unittest
+
+
+class BaseIso9660(unittest.TestCase):
+
+    """
+    Base class defining setup and tests for shared Iso9660 functionality
+    """
+
+    def setUp(self):
+        self.iso_path = os.path.abspath(os.path.join(os.path.dirname(__file__),
+                                                     os.path.pardir, ".data",
+                                                     "sample.iso"))
+        self.iso = None
+        self.tmpdir = tempfile.mkdtemp(prefix="avocado_" + __name__)
+
+    def basic_workflow(self):
+        """
+        Check the basic Iso9660 workflow
+        """
+        self.assertEqual(self.iso.read("file"),
+                         "file content\n")
+        dst = os.path.join(self.tmpdir, "file")
+        self.iso.copy(os.path.join("Dir", "in_dir_file"), dst)
+        self.assertEqual(open(dst).read(), "content of in-dir-file\n")
+        self.iso.close()
+        self.iso.close()    # check that double-close won't fail
+
+    @unittest.skipIf(os.getuid(), "This test is only available to root")
```

Comment:
I didn't want to use that as it's not good to mix original implementation and it's test. In rare cases this could generate false-negatives, but it should not false-positively skip the test, which is IMO better.